MobiMart/customer/views.py

```diff
@@ -9,6 +9,31 @@
 from .models import *
 
 # Create your views here.
+def checkpass(str):
+    is_digit=False
+    is_upper=False
+    is_lower=False
+    is_check=False
+    is_length=False
+    is_special=True
+    for i in str:
+        if i.isupper():
+            is_upper=True
+        if i.islower():
+            is_lower=True
+        if i.isdigit():
+            is_digit=True
+        if not i.isdigit() and not i.isalpha():
+            if i!='@' and i!='!' and i!='$':
+                is_special=False
+        is_check=is_digit and is_lower and is_upper and is_special
+        if(is_check):
+            break
```

Comment:
this condition must be removed because when all the conditions are satisfied, the loop breaks so it doesn't check anything after that. Think more and make it simple.

Reply:
Can you please elaborate this which condition you are saying.

Reply:
Like if passw1 = 'Heello@123_'
So here if you loop this, at position 7 i.e. 1 ; all the conditions are satisfied so the loop will break according to your code.
So _ is not checked.

Reply:
Ok understood